multiuser/keycloak/che-multiuser-keycloak-server/src/main/java/org/eclipse/che/multiuser/keycloak/server/deploy/KeycloakServletModule.java

```diff
@@ -32,7 +32,10 @@ protected void configureServlets() {
     filterRegex("^(?!.*(/docs/))(?!.*(/keycloak/settings/?|/api/oauth/callback/?)$).*")
         .through(IdentityIdLoggerFilter.class);
 
-    filterRegex("/user/?.*").through(UnavailableResourceInMultiUserFilter.class);
+    // Ban change password (POST /user/password) and create a user (POST /user/) methods
+    // but not remove user (DELETE /user/{USER_ID}
+    filterRegex("/user(/password|/)?").through(UnavailableResourceInMultiUserFilter.class);
```

Comment:
Isn't pattern should be `\/user(\/password|\/)?$` ? (extra $)

Reply:
Even more: `/user(/password/?|/)?$`

Reply:
I don't think that it will change a lot, but I agree that it makes sense to use `^` and `$` in regular expressions. So, do you think the following changes are OK
```
filterRegex("^/user(/password/?|/)?$")....
filterRegex("^/profile/.*/attributes$")....
```
?

Reply:
If you didn't indicate end of string,  it matches `/user/anything`  then. So delete won't work.

Reply:
Those are ok

Reply:
@mshaposhnik Your words make sense but I tested this PR and it works. So, it doesn't match `/user/anything`. And specifying `.*` the only way to make it work in this way.
let me double check this.

Reply:
So, thanks for pointing to `/password/?`
As about `^` and `$` they are really optional here since `com.google.inject.servlet.RegexUriPatternMatcher` uses `Matcher#matches` that 
```
Returns <tt>true</tt> if, and only if, the entire region sequence matches this matcher's pattern
```
while `/user/anything` contains region that matches to `/user(/password/?|/)?` but it is not that one which matches to the specified pattern